proto/public/api.proto

```diff
@@ -124,8 +124,68 @@ message ApiError {
   string error = 1;
 }
 
+message ResourceSpec {
+  string namespace = 1;
+  string resource = 2; // k8s resource e.g. deploy/foo, or 'all'
+}
+
+message ResourceSelection {
+  ResourceSpec spec = 1;
+  string label_selector = 2;
+}
+
+message ResourceError {
+  ResourceSpec spec = 1;
+  string error = 2;
+}
+
+message StatSummaryRequest {
+  ResourceSelection resource = 1;
+  TimeWindow time_window = 2;
+
+  oneof outbound {
+    Empty none = 3;
+    ResourceSpec out_to_resource   = 4;
+    ResourceSpec out_from_resource = 5;
```

Comment:
Based on my reading of the spec in #627, these two fields correspond to the `--out-from` and `--out-to` command line flags. In that case the naming makes sense, but I'm not sure I understand exactly what those flags do.

I'm guessing that, by default, the stat command returns inbound stats, but if either `--out-from` or `--out-to` is set, then the command instead returns outbound stats, filtered by either the source resource (`--out-from`) or the target resource (`--out-to`)? Whether or not we return inbound or outbound stats feels like a binary flag to me, whereas I would expect for us to be able to add filtering regardless of whether or not we're returning inbound or outbound stats. But it's also possible I'm not fully understanding the design.

On a related note, do we need this request message to also pass the values that are passed via the `--out-from-namespace` / `--out-from-selector` / `--out-to-namespace` / `--out-to-selector` flags?

I can post this feedback directly on #627 if that's more appropriate. Just figured I'd mention it while I was looking at the definition here.